lib/generators/shopify_app/home_controller/templates/index.html.erb

```diff
@@ -1,21 +1,71 @@
-<h2>Products</h2>
+<!DOCTYPE html>
+<html lang="<%= I18n.locale %>">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link
+      rel="stylesheet"
+      href="https://unpkg.com/@shopify/polaris@4.25.0/styles.min.css"
+      />
+<% if @with_session_token %>    <script>
+      document.addEventListener("DOMContentLoaded", async function() {
+        var SessionToken = window["app-bridge"].actions.SessionToken
+        var app = window.app;
 
-<ul>
-  <% @products.each do |product| %>
-    <li><%= link_to product.title, "https://#{@current_shopify_session.domain}/admin/products/#{product.id}", target: "_top" %></li>
-  <% end %>
-</ul>
+        app.dispatch(
+          SessionToken.request(),
+        );
 
-<hr>
+        // Save a session token for future requests
+        window.sessionToken = await new Promise((resolve) => {
+          app.subscribe(SessionToken.ActionType.RESPOND, (data) => {
+            resolve(data.sessionToken || "");
+          });
+        });
 
-<h2>Webhooks</h2>
+        var fetchProducts = function() {
+          var headers = new Headers({ "Authorization": "Bearer " + window.sessionToken });
+          return fetch("/products", { headers })
+            .then(response => response.json())
+            .then(data => {
+              var products = data.products;
 
-<% if @webhooks.present? %>
-  <ul>
-    <% @webhooks.each do |webhook| %>
-      <li><%= webhook.topic %> : <%= webhook.address %></li>
-    <% end %>
-  </ul>
-<% else %>
-  <p>This app has not created any webhooks for this Shop. Add webhooks to your ShopifyApp initializer if you need webhooks</p>
-<% end %>
+              if (products === undefined || products.length == 0) {
+                document.getElementById("products").innerHTML = "<br>No products to display.";
+              } else {
+                var list = "";
+                products.forEach((product) => {
+                  var link = `<a target="_top" href="https://<%%= @shop_origin %>/admin/products/${product.id}">`
```

Comment:
Why do we use the double-% here?

Reply:
The `template` method in home_controller_generator.rb essentially does two passes when it copies this file to its final destination: 

- The first time, it replaces all instances of `<%= %>` with their respective control flows. In this example, it conditionally executives the `<% if with_session_token %> <% end %>` flow to determine whether we want a vanilla javascript solution of an erb solution containing products and webhooks. 
- For it's second pass, it replaces all instances of `<%% erb code %>` with `<% erb code %>` to properly template this file when it's generated.

Reply:
So if `<% if with_session_token %>` passes, we get the following final file: https://gist.github.com/NabeelAhsen/0a134edf76c85a5a425e47836b2013bf

Otherwise, we get the following final index.html.erb file: https://gist.github.com/NabeelAhsen/a0269613246eb62df5c105c85a8aa4d9

Reply:
Got it, thanks for the detailed explanation 👍